### PR TITLE
Add checks to new item input

### LINKED ISSUE #10: 10. As a user, I want to be alerted when I’m entering an empty item or an item that's already on my list
## Summary
Users shouldn't be able to add an empty item to their list, or add the same item twice. If they try to do this, we need to show them an error message that explains the problem. This way, we’ll prevent some clutter in their lists.

## Acceptance criteria
- [x] Show an error message if the user tries to submit an empty item
- [x] Show an error message if the user tries to submit a new item that is _identical_ to an existing name. For instance, if the list contains `apples` and the user adds `apples`.
- [x] Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains `apples` and the user adds `aPples` or `apples,` or `a pples`.
- [x] The user’s original input is saved in the database

##  Notes
- There are a couple of ways to approach this task. If you’re not sure how to begin, _ask your mentors_ for advice!

### PR DESCRIPTION
**Pull request by etiry**
## Description

Adds checks to user input of new items to alert the user if they submitted an empty item name or submitted an item that is already on their list after normalizing case and punctuation

## Related Issue

closes #10 

## Acceptance Criteria

- [x] Show an error message if the user tries to submit an empty item
- [x] Show an error message if the user tries to submit a new item that is _identical_ to an existing name. For instance, if the list contains `apples` and the user adds `apples`.
- [x] Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains `apples` and the user adds `aPples` or `apples,` or `a pples`.
- [x] The user’s original input is saved in the database

## Type of Changes

`enhancement`

### Before

N/A

### After

Screenshot 2024-03-07 at 8 54 38 AM

Screenshot 2024-03-07 at 8 54 12 AM
3c18)

Screenshot 2024-03-07 at 8 55 17 AM

## Testing Steps / QA Criteria

- Sign in and select the dinner list
- Navigate to the Manage List view
- Try to submit the form without entering an item name
- Confirm that an alert displays saying "Please enter an item name"
- Try to add an item that is already on the list by entering it identically (e.g., Gruyere not gruyere)
- Confirm that an alert displays saying "This item already exists in the list"
- Try to add an item that is already on the list by changing the casing and/or adding punctuation or spaces (e.g., gruyere or Gruyere!)
- Confirm that an alert displays saying "This item already exists in the list"
- Try to add an item that is not on the list
- Confirm it gets added successfully
- Confirm it gets added as you typed it in rather than a normalized version (e.g., Oreos not oreos)

### CODE REVIEW

File: src/views/ManageList.jsx [hunk 1]
```diff
@@ -17,9 +17,34 @@ export function ManageList({ listPath, userId }) {
 		setShareEmail(target.value);
 	};
 
+	const currentList = data.map((item) => {
+		return item.name
+			.toLowerCase()
+			.replace(/[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~/\s]/g, '');
```
Comment (sdmisra) [inline src/views/ManageList.jsx:23]: praise: regular expressions are really intimidating to write at times, they way you both implemented it here looks like it'll catch any symbols that users might try to put in their list that might cause problems, nice!

**Review by sdmisra — DISMISSED** [review]
This looks good to me, nice work this week!

File: src/views/ManageList.jsx [hunk 2]
```diff
@@ -17,9 +17,34 @@ export function ManageList({ listPath, userId }) {
...
+			if (currentList.includes(itemName)) {
+				alert('This item already exists in the list.');
+				return;
+			}
+		}
+
+		if (!newItem.itemName) {
+			alert('Please enter an item name');
+			return;
+		}
```
Comment (raaynaldo) [inline src/views/ManageList.jsx]: refactor: we could move the second if clause to be early, so we don't need to have nested if for the first if clause

```suggestion
		if (!newItem.itemName) {
			alert('Please enter an item name');
			return;
		}

		//edge case: what if someone decides to put only puncutation as item name
		const itemName = newItem.itemName
			.toLowerCase()
			.replace(/[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~/\s]/g, '');
		console.log('itemName:', itemName);

		if (currentList.includes(itemName)) {
			alert('This item already exists in the list.');
			return;
		}
```

**Review by raaynaldo — CHANGES_REQUESTED** [review]

**Comment by raaynaldo** [comment]
Great job on utilizing regex

**Review by DevinaG007 — DISMISSED** [review]
Praise: Nice comparison choice used on line 36 of ```ManageList```. I tested "New England Clam Chowder" versus "Clam Chowder" and it worked! 

success

**Review by raaynaldo — APPROVED** [review]
Thanks!

**Pull request merged**